helm-chart/todoapp/templates/pvc.yml

```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ .Chart.Name }}-pvc-data
+  namespace: {{ .Values.namespace }}
+spec:
+  volumeName: {{ .Chart.Name }}-pv-data
```

Comment:
The `volumeName` field is set to a specific value. Ensure that a PersistentVolume with this name already exists, as specifying a `volumeName` binds the PVC to an existing PV. If this is not intended, consider removing this field to allow dynamic provisioning.